src/model/Model_Currency.cpp

```diff
@@ -203,59 +204,80 @@ const wxString Model_Currency::toString(double value, const Data* currency, int
         }
     }
 
-    if (defaultLocaleSupport.empty())
+    if (d.empty())
     {
         try {
-            defaultLocaleSupport = "Y";
-            fmt::format(std::locale("en_US"), "{:L}", 123);
+            fmt::format(std::locale("en_US.UTF-8"), "{:L}", 123);
+            d = "Y";
         }
         catch (...) {
-            defaultLocaleSupport = "N";
+            d = "N";
         }
     }
 
-    if (decimal.empty()) {
-        if (use_locale == "Y")
-            decimal = wxString(fmt::format(std::locale(locale.c_str()), "{:L}", 1.1)).Mid(1, 1);
-        else
-            decimal = ".";
-    }
-
     if (precision < 0) {
         precision = log10(currency ? currency->SCALE : GetBaseCurrency()->SCALE);
     }
 
-    int k = pow10(precision);
-    long double v = value * k;
-    v = round(v) / k;
+    auto l = (use_locale == "Y" ? std::locale(locale.c_str()) : std::locale("en_US.UTF-8"));
+    std::string s;
+    value += LIMIT; //to ignore the negative sign on values of zero #564
 
-    wxString s;
-    if (defaultLocaleSupport == "Y")
+    switch (precision)
     {
-        s = fmt::format((use_locale == "Y" ? std::locale(locale.c_str()) : std::locale("en_US"))
-            , "{:L}", static_cast<unsigned long long>(fabs(v) + 5 / (pow10(precision + 1))));
-    }
-    else {
-        //In case if no en_US supported don't use any one
-        s = fmt::format("{:d}", static_cast<unsigned long long>(fabs(v) + 5 / (pow10(precision + 1))));
+    case (0):
+        s = d == "Y" ? fmt::format(l, "{:.0Lf}", value) : fmt::format("{:.0f}", value);
```

Comment:
Would be more readable as:  
s = (d == "Y") ? fmt::format(l, "{:.0Lf}", value) : fmt::format("{:.0f}", value);